src/components/MatchPage/Match.jsx

```diff
@@ -1,31 +1,158 @@
+                userName2[1] = target.value; // 온전한 이름 저장
+            }
+        } else {
+            // 온전한 한글이 아닐때
+            if (target.classList[2] === FIRSTNAME) {
+                // 첫 번재 이름이 한글이 아닐때
+                setIsErr1(true);
+            } else if (target.classList[2] === 'secondName') {
+                // 두 번재 이름이 한글이 아닐때
+                setIsErr2(true);
+            }
+        }
+    };
+
+    const beforeLocation = (e) => {
+        //submit
+        if (
+            // 예외처리
+            userName1[0] !== undefined &&
+            userName2[0] !== undefined &&
+            !isErr1 &&
+            !isErr2 &&
+            userName1[1].length === 3 &&
+            userName2[1].length === 3
+        ) {
+            names[0] = userName1[0];
+            names[1] = userName2[0];
+            names[2] = userName1[1];
+            names[3] = userName2[1];
+            setIsClicked(true); // 예외처리가 되면 로딩 컴포넌트 띄우기
+        } else {
+            try {
+                // 엔터키로 함수를 호출하면 오류가 남
+                e.preventDefault(); // 페이지 이동 막기
+            } catch {
+                isOnErr = true; // 에러 확인
+            }
+            if (
+                // 첫 번째 이름에서 오류가 났을 경우
+                userName1[0]?.length === undefined ||
+                userName1[1]?.length === undefined ||
+                userName1[1].length !== 3
+            ) {
+                setIsErr1(true);
+            }
+            if (
+                // 두 번째 이름에서 오류가 났을 경우
+                userName2[0]?.length === undefined ||
+                userName2[1]?.length === undefined ||
+                userName2[1].length !== 3
+            ) {
+                setIsErr2(true);
+            }
+        }
+    };
+
+    const onEnter = () => {
+        // 엔터키로 제출했을 경우
+        beforeLocation(); // 오류 확인
+        if (!isOnErr) {
+            // 오류가 안났으면?
+            navigate('/match/result', {
+                // 페이지 이동과 함께 이름들을 보낸다
+                state: {
+                    names: names,
+                },
+            });
+        }
+    };
+
     return (
-        <S.Container>
-            <S.VioletText>두 분의 이름을 입력해주세요</S.VioletText>
-            <S.NameContainer>
-                <div>
-                    <S.NameInput
-                        placeholder="이름"
-                        color={isErr ? '#FF0000' : 'white'}
-                    ></S.NameInput>
-                    <S.ErrMessege>{isErr ? '에러메세지' : ''}</S.ErrMessege>
-                </div>
-                <div>
-                    <S.NameInput
-                        placeholder="이름"
-                        color={isErr ? '#FF0000' : 'white'}
-                    ></S.NameInput>
-                    <S.ErrMessege>{isErr ? '에러메세지' : ''}</S.ErrMessege>
-                </div>
-            </S.NameContainer>
-            <S.TextContainer>궁합 보기</S.TextContainer>
-        </S.Container>
+        <div>
+            {isClicked ? <Loading></Loading> : null}
```

Comment:
```suggestion
            {isClicked && <Loading></Loading>}
```

위 코드의 경우 조건이 거짓일 때, return 하는 값이 null이므로
삼항 연산자 보다 `&&` AND 연산자를 사용하는게 더 나아보이네요

Reply:
수정 했습니다